src/is-package-changed.ts

```diff
@@ -61,15 +61,18 @@ async function isPackageChanged(
     };
 
     if (callback) {
+        if (lockfile) {
+            // hash may have changed since package-lock.file could have been updated after command
+            result.hash = `${getPackageHash(packagePath)}${
+                getPackagelockHash(packagelockPath) ?? ''
+            }`;
+        }
```

Comment:
(install) command has not been executed here yet, so the hash will not have been changed. I believe this code needs to run after line 71 and only for non CI environments.
```
let canWriteHash = await callback(result);
```

Reply:
Also, on line 82 we expose a function that writes the hash on your file system. I think that we need to recalculate the hash there as well in case of `lockfile=true`